lib/index.js

```diff
@@ -495,6 +499,40 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
   }
 };
 
+/**
+ * Deletes any state persisted by cross domain analytics.
+ * * seg_xid (and metadata) from cookies
+ * * crossDomainId from traits in localStorage
+ *
+ * The deletion logic is run only if deletion is enabled for this project, and
+ * when either the seg_xid cookie or crossDomainId localStorage trait exists.
+ *
+ * @api private
+ */
+Segment.prototype.deleteCrossDomainIdIfNeeded = function() {
+  // Only continue if deletion is enabled for this project.
+  if (!this.options.deleteCrossDomainId) {
+    return;
+  }
+
+  // Only continue if we have any lingering identifiers.
+  if (this.cookie('seg_xid') == null && this.analytics.user().traits().crossDomainId == null) {
```

Comment:
Is it necessary to account for a case where `seg_xid`  is empty? I'm guessing there is a slim to none chance this occurs, but just checking. 

What about if the other identifiers such as `seg_xid_fd` and `seg_xid_ts` are null/empty? 

The only think I wonder about is in the case where this is reenabled for Fox, if having lingering _other_ identifiers / empty values would impact the re-enablement of the XID Service.

Reply:
This checks if both the cookie and localStorage trait don't exist. If both don't exist, then we don't need to do anything can bail. Otherwise at least 1 exists, and we proceed to delete both.

One improvement we could make is make the checks independent (i.e. delete cookie if it exists and delete localStorage trait if it exists).

we are not checking here if the other metadata (e.g. seg_xid_fd or seg_xid_ts) are empty, so it doesn't really matter in this case.

Reply:
> One improvement we could make is make the checks independent (i.e. delete cookie if it exists and delete localStorage trait if it exists).

Updated now!